image-hd.c

```diff
@@ -626,6 +626,14 @@ static int hdimage_setup(struct image *image, cfg_t *cfg)
 	hd->fill = cfg_getbool(cfg, "fill");
 	hd->disk_uuid = cfg_getstr(cfg, "disk-uuid");
 
+	if (!image->size && is_block_device(imageoutfile(image))) {
```

Comment:
Hmm, I think we should either require that `image->size` is not set for block devices or check that it is less or equal than the real size. I don't really see a use-case where the partition table does not handle the full device size, so I think I would prefer the first on to avoid surprises. Especially for GPT (where this makes a difference) having the secondary table anywhere but the end of the device is not a good idea anyways.

Reply:
Good point re GPT. Yes, let's just enforce that image->size is auto-computed for block device targets. That can be relaxed later (replaced by a size <= real size check && error if ->gpt and not ==) if somebody comes up with a use case.

Reply:
Fixed, I think.